### PR TITLE
Add preserve stop times flag to avoid normalizing stop_sequence

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds a `FeedSource#preserveStopTimes` flag to avoid the stop_sequence normalization that typically happens when making a snapshot for the editor.

re conveyal/gtfs-lib#283. 

**Note:** needs `pom.xml` updated to gtfs-lib@6.0.2 in order for build to pass (conveyal/gtfs-lib#285 needs to be merged and released first).

Note: there is no corresponding datatools-ui PR to set this flag (primarily just so that users don't "try this at home"). It must be done with a MongoDB update statement like: 
```json
db.getCollection('FeedSource').update({"_id": "YOUR_TEST_FEED_ID"}, {"preserveStopTimes": true})
```

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/editor/jobs/ExportSnapshotToGTFSJob.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public void jobLogic() {
         FeedLoadResult result = exporter.exportTables();
         if (result.fatalException != null) {
             status.fail(String.format("Error (%s) encountered while exporting database tables.", result.fatalException));
+            return;
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/editor/jobs/ExportSnapshotToGTFSJob.java:55]: Should we delete `tempFile` before returning?

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 2]
```diff
@@ -47,9 +47,17 @@
...
     public String projectId;
 
+    /**
+     * When snapshotting a GTFS feed for editing, gtfs-lib currently defaults to normalize stop sequence values to be
+     * zero-based and incrementing. This can muck with GTFS files that are linked to GTFS-rt feeds by stop_sequence, so
+     * this override flag currently provides a workaround for feeds that need to be edited but do not need to edit
+     * stop_times or individual patterns. WARNING: enabling this flag for a feed and then attempting to edit patterns in
+     * complicated ways (e.g., modifying the order of pattern stops) could have unexpected consequences.
+     */
+    public boolean preserveStopTimes;
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:59]: Rename to `preserveStopTimesSequence` and simplify the comment above.

**Comment by landonreed** [comment]
OK, just waiting on the gtfs-lib release. I'll merge this once I've updated the pom.

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 3]
```diff
@@ -47,9 +47,17 @@
...
      */
-    //@JsonView(JsonViews.DataDump.class)
     public String projectId;
 
+    /**
+     * When snapshotting a GTFS feed for editing, gtfs-lib currently defaults to normalize stop sequence values to be
+     * zero-based and incrementing. This can muck with GTFS files that are linked to GTFS-rt feeds by stop_sequence, so
+     * this override flag currently provides a workaround for feeds that need to be edited but do not need to edit
+     * stop_times or individual patterns. WARNING: enabling this flag for a feed and then attempting to edit patterns in
+     * complicated ways (e.g., modifying the order of pattern stops) could have unexpected consequences.
```
Comment (evansiroky) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:57]: Seems like it'd be good to have some kind of UI warning for this.

Comment (evansiroky) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java:57]: oh, just read the PR description. Maybe edit this comment to say that there is no UI setting for this and it is not recommended to do this unless absolutely necessary.

**Pull request merged**